### PR TITLE
Automatically sort string lists within complex statements

### PR DESCRIPTION
**Pull request by vladmos**
Buildifier currently sorts some string lists (such as `srcs` or `deps` arguments) automatically, but it doesn't sort anything if the value is not a list but a complex expression that contains valid sortable lists inside (such as list + constant, or a select statement with inner lists). This commit fixes that behaviour.

The change doesn't introduce any new sortable parameters, it only touches the parameters that would have been sorted anyway if they were plain string lists.

The change is not backwards compatible and will require a global cleanup on the existing files.

### CODE REVIEW

File: build/rewrite.go [hunk 1]
```diff
@@ -448,12 +448,12 @@ func sortStringLists(f *File, w *Rewriter) {
 					continue
 				}
 				if w.IsSortableListArg[key.Name] ||
-					w.SortableAllowlist[context] ||
-					(!disabled("unsafesort") && allowedSort(context)) {
+						w.SortableAllowlist[context] ||
```
Comment (meisterT) [inline build/rewrite.go:451]: nit: indentation looks broken

File: build/rewrite.go [hunk 2]
```diff
@@ -560,6 +560,37 @@ func SortStringList(x Expr) {
 	list.List = sortStringExprs(list.List)
 }
 
+// FindAndSortStringLists finds and sorts string lists recursively withing
```
Comment (meisterT) [inline build/rewrite.go:563]: nit: within

File: build/rewrite.go (hunk 1 above)
Comment (vladmos) [reply build/rewrite.go]: Fixed with gofmt

File: build/rewrite.go (hunk 2 above)
Comment (vladmos) [reply build/rewrite.go]: Done.

**Review by meisterT — APPROVED** [review]

**Review by laurentlb — APPROVED** [review]
lgtm

**Pull request merged**